### DISCUSSION TITLE
Can autoPlacement use "most space" strategy for alignment too?

### DISCUSSION THREAD

**Discussion by krutoo** [floating-ui/floating-ui#2155] [Q&A] [answered]
Hi, i use `autoPlacement` and it works perfectly.

But i want to apply "most space" strategy to alignment (start/end) too:

alignment should be chosen for most space side.

Here is an example:
https://codesandbox.io/s/can-autoplacement-use-most-space-strategy-for-alignment-too-iiotvw?file=/src/styles.css 

How can i do it?

**Comment by atomiks** [comment]
I think it's related (or is) this bug: https://github.com/floating-ui/floating-ui/issues/1636

**Reply by krutoo** [reply]
maybe this is a related issue but i'm not sure

I just want my dropdown (which is wider than its _reference_) to be aligned horizontally like this:

- if there is more space to the left of the _reference_ than to the right, then the dropdown is aligned to the right side of the _reference_
- if there is more space to the right of the _reference_ than to the left, then the dropdown is aligned to the left side of the _reference_

**Reply by atomiks** [reply]
Even though, if fixed, it would check the most space based on the floating element, it should still achieve the same thing as what you want with the reference. But it's currently a bug, so it won't work without a fix. As a workaround, you can create your own middleware which achieves the same thing, or wait for/file a PR to fix the issue in `autoPlacement`.

**Reply by atomiks** [reply, marked as answer]
This is now fixed in `v1.2.0` of the `core`/`dom` libs by passing `crossAxis: true` to `autoPlacement`:

https://codesandbox.io/s/can-autoplacement-use-most-space-strategy-for-alignment-too-forked-l9evyr?file=/src/App.js

**Reply by krutoo** [reply]
@atomiks Awesome! Thank you a lot
